include/extractor/guidance/intersection.hpp

```diff
@@ -204,6 +212,15 @@ template <typename Self> struct EnableIntersectionOps
     // Returns the number of roads we can not enter at this intersection, respectively.
     auto countNonEnterable() const { return self()->size() - self()->countEnterable(); }
 
+    // check if all roads between begin and end allow entry
+    template <typename iteratorT>
+    bool hasAllValidEntries(const iteratorT begin, const iteratorT end) const
+    {
+        BOOST_ASSERT(end < self()->end());
+
+        return all_of(begin, end, [](const ConnectedRoad &road) { return road.entry_allowed; });
```

Comment:
this should use type `auto`, since we can also call it on an intersectionView which does not have a connected road.